spark3/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -131,28 +138,52 @@ public SparkScanBuilder caseSensitive(boolean isCaseSensitive) {
 
   @Override
   public void pruneColumns(StructType requestedSchema) {
-    this.requestedProjection = requestedSchema;
+    this.requestedProjection = new StructType(Stream.of(requestedSchema.fields())
+        .filter(field -> MetadataColumns.nonMetadataColumn(field.name()))
+        .toArray(StructField[]::new));
+
+    Stream.of(requestedSchema.fields())
```

Comment:
Are there any cases when `pruneColumns` is going to be called multiple times? Should we worry about it at all?

Reply:
I see we call distinct in `schemaWithMetadataColumns`, never mind.